src/blockdata/opcodes.rs

```diff
@@ -803,11 +817,19 @@ macro_rules! ordinary_opcode {
     ($($op:ident),*) => (
         #[repr(u8)]
         #[doc(hidden)]
-        #[derive(Copy, Clone, PartialEq, Eq, Debug)]
+        #[derive(Copy, Clone, PartialEq, Eq)]
         pub enum Ordinary {
             $( $op = all::$op.code ),*
         }
 
+        impl fmt::Display for Ordinary {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                match *self {
+                    $(Ordinary::$op => {write!(f, "{}", all::$op)}),*
```

Comment:
`f.write_str(stringify!(op))` may have a bit better performance. But just as this one it ignores alignment and width. You don't have to worry about it too much but if you'd like to do add the feature I'd be happy.

Reply:
I hope i fully understood your comment. I used _pad_ and _format!_  to make the Display aware of alignment and width.